components/automate-ui/src/app/pages/project/rules/project-rules.component.ts

```diff
@@ -131,7 +130,9 @@ export class ProjectRulesComponent implements OnInit, OnDestroy {
 
     this.ruleForm = this.fb.group({
       // Must stay in sync with error checks in project-rules.component.html
-      name: ['', [Validators.required, Validators.pattern(Regex.patterns.NON_BLANK)]],
+      name: [
+        this.rule.name || '',
```

Comment:
This is odd. Why do we need this?

Reply:
Since this component is used for updating and creating, the value needs to either start as a blank string or the existing rule name. This was confusing so we changed the form to initialize as a create form or an update form depending on the value of `editingRule`